src/main/java/com/project/portfolio/springboot/onlinesellingandshopping/serverapp/security/ApplicationUserDetailsService.java

```diff
@@ -0,0 +1,38 @@
+package com.project.portfolio.springboot.onlinesellingandshopping.serverapp.security;
+
+import com.project.portfolio.springboot.onlinesellingandshopping.serverapp.module.AuthGroup;
+import com.project.portfolio.springboot.onlinesellingandshopping.serverapp.module.User;
+import com.project.portfolio.springboot.onlinesellingandshopping.serverapp.repository.AuthGroupRepository;
+import com.project.portfolio.springboot.onlinesellingandshopping.serverapp.repository.UserRepository;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class ApplicationUserDetailsService implements UserDetailsService {
+
+    private final UserRepository userRepository;
+    private final AuthGroupRepository authGroupRepository;
+
+    public ApplicationUserDetailsService(UserRepository userRepository, AuthGroupRepository authGroupRepository) {
+        this.userRepository = userRepository;
+        this.authGroupRepository = authGroupRepository;
+    }
+
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = (User) this.userRepository.findByUsername(username).get();
```

Comment:
Assuming the return value is of type Optional<User> ? In this case you want to check it's not empty first before retrieving the value.